### PR TITLE
replace strncpy

### PR DESCRIPTION
**Pull request by nichtsfrei**
- Refactor: ignore -Wstringop-truncation due to FOTIFY=2
- Refactor: remove -Werror
- Refactor: enable -Werror replace strncpy with memcpy

### CODE REVIEW

File: base/prefs.c [hunk 1]
```diff
@@ -174,7 +174,8 @@ prefs_config (const char *config)
   if (!global_prefs)
     prefs_init ();
 
-  strncpy (buffer, config, sizeof (buffer));
+  memset (buffer, 0, sizeof (buffer));
+  memcpy (buffer, config, sizeof (buffer));
```
Comment (y0urself) [inline base/prefs.c:178]:
```suggestion
  memset (buffer, 0, sizeof (buffer));
  memcpy (buffer, config, sizeof (buffer) - 1);
```

we might want to have a null termination at the end?

I have seen in the other places the usage of `bzero` ... we could use it here too?

Comment (nichtsfrei) [reply base/prefs.c:178]: I would rather use the standard memset tbh

File: osp/osp.c [hunk 2]
```diff
@@ -152,7 +152,8 @@ osp_connection_new (const char *host, int port, const char *cacert,
         }
 
       addr.sun_family = AF_UNIX;
-      strncpy (addr.sun_path, host, sizeof (addr.sun_path) - 1);
+      memset (addr.sun_path, 0, sizeof (addr.sun_path) - 1);
```
Comment (y0urself) [inline osp/osp.c:155]:
```suggestion
      memset (addr.sun_path, 0, sizeof (addr.sun_path));
```

I guess we want to initialize everything?

**Review by y0urself — APPROVED** [review]
changes look good to me

**Pull request merged**